### ISSUE TITLE
A more generic "File Gateway" would be the ideal solution here

### ISSUE THREAD

**Issue by zbruhnke** [moov-io/achgateway#111]
In this case we could use the same setup for uploading wire files, ach files, BSA/Compliance files as needed to partners, you could still have different upload agents for each if needed but the generic nature of this need would go a long way.

I feel like otherwise we'll end up duplicating several parts of ach gateway multiple times with various systems when some configuration changes would be everything that is needed, though I understand that would also mean a fairly good size refactor of the codebase possibly.

**Comment by adamdecaf** [comment]
I agree that a more generic solution would help with Wires, ICL, generic reports, etc and that's a personal goal of mine. 

Converting achgateway into a generic uploader might be more work than I'm willing to signup for. It's working pretty well so far and has some ACH details backed into it. We know [that the config is confusing](https://github.com/moov-io/achgateway/issues/44) in several spots, so a cleanup there would be nice. 

We're talking about a "file collector" internally which is a generic downloader from multiple SFTP endpoints, so a generic uploader would be very similar to that. cc @atonks2 and @alovak
